File: app/src/main/java/com/example/smarthome/user/services/IUserRetrofitService.java
package com.example.smarthome.user.services;

import com.example.smarthome.user.models.LoginResponse;
import com.example.smarthome.user.models.SigninBody;
import com.example.smarthome.user.models.User;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface IUserRetrofitService {

    @POST("/user/signup")
    Call<String> signup(@Body User user);

    @POST("/user/signin")
    Call<ResponseBody> signin(@Body SigninBody signinBody);

}

File: app/src/main/java/com/example/smarthome/user/services/IUserRetrofitService.kt
package com.example.smarthome.user.services

import com.example.smarthome.user.models.LoginRequest
import com.example.smarthome.user.models.User
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface IUserRetrofitService {
    @POST("/user/signup")
    fun signup(@Body user: User) : Call<ResponseBody>

    @POST("/user/signin")
    fun signin(@Body loginRequest: LoginRequest) : Call<ResponseBody>

    @GET("/server")
    fun getStatus(): Call<ResponseBody>
}